File: Cpp/477_Total_Hamming_Distance.cpp
class Solution {
public:
    int totalHammingDistance(vector<int>& nums) {
        int result = 0;
        
        for(int i = 0; i < 32; ++i) {
            int ones = 0;
            for(int item : nums) {
                if(item & (1 << i)) ++ones;
            }
            result += ones * (nums.size() - ones);
        }
        
        return result;
    }
};

File: Kotlin/477_Total_Hamming_Distance.kt
class Solution {
    fun totalHammingDistance(nums: IntArray): Int {
        var result = 0
        
        for (i in 0 until 32) {
            var ones = nums.count{ it.and(1.shl(i)) > 0 }
            result += ones * (nums.size - ones)
        }

        return result
    }
}